lib/dependencies/AMDDefineDependency.js

```diff
@@ -2,113 +2,137 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var NullDependency = require("./NullDependency");
+"use strict";
+const NullDependency = require("./NullDependency");
 
-function AMDDefineDependency(range, arrayRange, functionRange, objectRange) {
-	NullDependency.call(this);
-	this.range = range;
-	this.arrayRange = arrayRange;
-	this.functionRange = functionRange;
-	this.objectRange = objectRange;
+class AMDDefineDependency extends NullDependency {
+	constructor(range, arrayRange, functionRange, objectRange) {
+		super();
+		this.range = range;
+		this.arrayRange = arrayRange;
+		this.functionRange = functionRange;
+		this.objectRange = objectRange;
+	}
+
+	get type() {
+		return "amd define";
+	}
 }
-module.exports = AMDDefineDependency;
 
-AMDDefineDependency.prototype = Object.create(NullDependency.prototype);
-AMDDefineDependency.prototype.constructor = AMDDefineDependency;
-AMDDefineDependency.prototype.type = "amd define";
+AMDDefineDependency.Template = class AMDDefineDependencyTemplate {
+	get definitions() {
+		return {
```

Comment:
What do you mean by ES5 solution?
the current implementation also creates this object on each call, to prevent that having it in the const somewhere would probably make sense.
the probably "fastest" lookup would be to set it as an attribute of the class as in
`AMDDefineDependencyTemplate.definitions = ...`
which would again open to door to assigning values to the class after its definition, which I would be ok with, but they still look weird :)
so would a "hybrid" solution be ok? as in:
```
const definitions = ...;
class AMDDefineDependencyTemplate {
...
get definitions() {
  return definitions;
}
```

that way its easy to switch to:
```
class AMDDefineDependencyTemplate {
   static definitions = ...;
}
```
once they are in the spec (should they become a part of the spec)